dds/DCPS/TypeSupportImpl.cpp

```diff
@@ -84,7 +84,11 @@ void TypeSupportImpl::populate_dependencies(const RcHandle<XTypes::TypeLookupSer
       const size_t tobj_size = serialized_size(XTypes::get_typeobject_encoding(), iter->second);
       XTypes::TypeIdentifierWithSize tmp = {*it, static_cast<ACE_CDR::ULong>(tobj_size)};
       deps_with_size.append(tmp);
-    } else if (DCPS_debug_level >= 1) {
+    } else if ((DCPS_debug_level >= 1) &&
+               (it->kind() ==  XTypes::TI_STRONGLY_CONNECTED_COMPONENT ||
+                it->kind() == XTypes::EK_COMPLETE ||
+                it->kind() == XTypes::EK_MINIMAL ||
+                it->kind() == XTypes::TK_ANNOTATION)) {
```

Comment:
```suggestion
               !XTypes::is_fully_descriptive(*it)) {
```

Reply:
I thought I had written that code somewhere.

Reply:
Neat

Reply:
I just looked at the function and the spec definition for fully descriptive TypeIdentifiers. I think using this function can lead to the warning message printed for non-fully-descriptive plain collection types, but plain collection types don't have TypeObjects regardless of whether they are fully descriptive or not.

Reply:
So should I change it to full descriptive or are we worried about what Son mentioned?